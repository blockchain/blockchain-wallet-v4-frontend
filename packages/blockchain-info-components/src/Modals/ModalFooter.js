import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.div`
  position: relative;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  padding: 20px 30px;
  box-sizing: border-box;
  border-top: 1px solid ${props => props.theme['gray-1']};
`
const Footer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: ${props => props.alignment};
  align-items: center;
  width: 100%;
`

const selectAlignment = align => {
  switch (align) {
    case 'left':
      return 'flex-start'
    case 'right':
      return 'flex-end'
    case 'center':
      return 'center'
    case 'spaced':
      return 'space-between'
    default:
      return ''
  }
}

const ModalFooter = props => {
  const { align, children, ...rest } = props
  const alignment = selectAlignment(align)
  return (
    <Wrapper {...rest}>
      <Footer alignment={alignment}>{children}</Footer>
    </Wrapper>
  )
}

ModalFooter.propTypes = {
  align: PropTypes.oneOf(['left', 'right', 'center', 'spaced'])
}

ModalFooter.defaultProps = {
  align: 'left'
}

export default ModalFooter
