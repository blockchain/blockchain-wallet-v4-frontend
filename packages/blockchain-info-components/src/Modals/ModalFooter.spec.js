import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import ModalFooter from './ModalFooter'

describe('ModalFooter component', () => {
  it('default renders correctly', () => {
    const component = shallow(<ModalFooter align='left' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
