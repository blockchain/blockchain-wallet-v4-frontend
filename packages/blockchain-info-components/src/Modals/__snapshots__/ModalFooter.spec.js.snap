// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalFooter component default renders correctly 1`] = `
<ModalFooter__Wrapper>
  <ModalFooter__Footer
    alignment="flex-start"
  />
</ModalFooter__Wrapper>
`;
